src/meta/Intersection.ts

```diff
@@ -0,0 +1,167 @@
+import global from '@dojo/shim/global';
+import WeakMap from '@dojo/shim/WeakMap';
+import { Base } from './Base';
+
+interface IntersectionDetail {
+	entries: WeakMap<Element, IntersectionObserverEntry>;
+	intersectionObserver?: IntersectionObserver; // attached observer
+	options: IntersectionGetOptions;
+	root: string;
+	rootMargin: string | undefined;
+	thresholds: number[]; // thresholds the observe should be attached with
+}
+
+export interface IntersectionGetOptions {
+	root?: string;
+	rootMargin?: string;
+	threshold?: number;
+	thresholds?: number[];
+}
+
+export interface IntersectionResult {
+	intersectionRatio: number;
+	isIntersecting: boolean;
+}
+
+const defaultIntersection: IntersectionResult = Object.freeze({
+	intersectionRatio: 0,
+	isIntersecting: false
+});
+
+export class Intersection extends Base {
+	private _details: IntersectionDetail[] = [];
+
+	private _getDetails(options: IntersectionGetOptions): IntersectionDetail {
+		let {
+			root = '',
+			rootMargin,
+			threshold,
+			thresholds = []
+		} = options;
+		const {
+			_details: details
+		} = this;
+		if (typeof threshold === 'number' && !thresholds.length) {
```

Comment:
We wouldn't need this `thresholds`, is typed as `number | number[]`. 

Could just do:

```ts
thresholds = Array.isArray(thresholds) ? thresholds : [ thresholds ];
```